<?xml version="1.0" ?>
<launch>
    <arg name="input_camera_name" default=""/>
    <arg name="output_camera_name" default=""/>
    <arg name="yolact_edge_weights" default="/root/yolact_edge/weights/yolact_edge_resnet50_54_800000.pth"/>
    <arg name="score_threshold" default="0.6" />
    <arg name="disable_tensorrt" default="False"/>
    <arg name="use_intel_camera" default="False"/>


    <node unless="$(arg use_intel_camera)" name="yolact_edge_ros" pkg="yolact_edge_ros" type="yolact_edge_ros.py" respawn="false" output="screen" >
        <param name="input_camera_name" value="$(arg input_camera_name)" />
        <param name="output_camera_name" value="$(arg output_camera_name)" />
        <param name="yolact_edge_weights" value="$(arg yolact_edge_weights)" />
        <param name="score_threshold" value="$(arg score_threshold)" />
        <param name="disable_tensorrt" value="$(arg disable_tensorrt)" />
    </node>

    <node if="$(arg use_intel_camera)" name="yolact_edge_ros" pkg="yolact_edge_ros" type="yolact_edge_intel.py" respawn="false" output="screen" >
        <param name="output_camera_name" value="$(arg output_camera_name)" />
        <param name="yolact_edge_weights" value="$(arg yolact_edge_weights)" />
        <param name="score_threshold" value="$(arg score_threshold)" />
        <param name="disable_tensorrt" value="$(arg disable_tensorrt)" />
    </node>


</launch>